Source/Dafny/AST/DafnyAst.cs

```diff
@@ -9070,6 +9086,48 @@ public abstract class Expression {
       get { yield break; }
     }
 
+    private IToken startTok;
+    private IToken endTok;
+
+    // Creates a token on the entire range of the expression.
+    // Used only for error reporting.
+    public IToken RangeToken => new RangeToken(StartTok, EndTok);
+
+    // Computes the start and end token of this expression
+    private void RecomputeStartEnd() {
+      startTok = tok;
+      endTok = tok;
+      foreach (var e in SubExpressions) {
+        if (e.tok.pos < startTok.pos) {
+          startTok = e.tok;
```

Comment:
Shouldn't this be `e.StartTok` ?

Reply:
Yes that would be even better, thanks. I'm also adding a check that the filename should be the same.